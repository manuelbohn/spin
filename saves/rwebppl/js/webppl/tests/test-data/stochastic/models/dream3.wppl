var guideBool = function(x) {
  var p = param({name: x ? 'a' : 'b'});
  return Bernoulli({p: Math.sigmoid(p)});
};

var model = function() {
  var arr = mapData({data: [false]}, function(x) {
    var bool = sample(Bernoulli({p: .5}), {guide() { return guideBool(x); }});
    observe(Bernoulli({p: bool ? .1 : .9}), x);
    return bool;
  });
  // We need this check because mapData returns undefined while
  // dreaming.
  return arr && first(arr);
};
