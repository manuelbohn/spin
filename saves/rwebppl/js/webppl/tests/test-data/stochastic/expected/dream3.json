{
  "hist": {
    "true": 0.9,
    "false": 0.1
  }
}
